pkg/api/validation/events.go

```diff
@@ -25,13 +29,35 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	// TODO: There is no namespace required for node.
-	if event.InvolvedObject.Kind != "Node" &&
-		event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
+
+	nonNamespacedKindFlag, err := isNonNamespacedKind(event.InvolvedObject.Kind, event.InvolvedObject.APIVersion)
+	if err != nil {
+		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.Kind", event.InvolvedObject.Kind, fmt.Sprintf("couldn't check whether namespace is allowed: %s", err)))
```

Comment:
On Oct 29, 2015, at 12:36 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/validation/events.go
<https://github.com/kubernetes/kubernetes/pull/16146#discussion_r43409922>:

> @@ -25,13 +29,35 @@ import (
>  // ValidateEvent makes sure that the event makes sense.
>  func ValidateEvent(event *api.Event) errs.ValidationErrorList {
>  	allErrs := errs.ValidationErrorList{}
> -	// TODO: There is no namespace required for node.
> -	if event.InvolvedObject.Kind != "Node" &&
> -		event.Namespace != event.InvolvedObject.Namespace {
> -		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
> +
> +	nonNamespacedKindFlag, err := isNonNamespacedKind(event.InvolvedObject.Kind, event.InvolvedObject.APIVersion)
> +	if err != nil {
> +		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.Kind", event.InvolvedObject.Kind, fmt.Sprintf("couldn't check whether namespace is allowed: %s", err)))

Hm. What about events about objects from 3rd parties? @smarterclayton
<https://github.com/smarterclayton> does OpenShift make k8s events about
its own objects?


Yes.  In our groups and versions.


(@feihujiang <https://github.com/feihujiang> I'm sorry I keep making this
so complicated!)

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16146/files#r43409922>.